File: scala/5.scala
def gcd(a: Int, b: Int): Int = if (b == 0) a else gcd(b, a % b)

def lcm(a: Int, b: Int): Int =  a * (b / gcd(a, b))

val result = (1 to 20).foldLeft(20)(lcm(_, _))

println(result)
assert(result == 232792560)

File: ruby/5.rb
def gcd(a, b)
  if b == 0 then a else gcd(b, a % b) end
end

def lcm(a, b)
  a * (b / gcd(a, b))
end

result = (1..20).inject(20) { |a, b| lcm(a, b) }

puts result
raise Error unless result == 232792560
